File: leetcode/242_easy_valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length())
            return false;
        unordered_map<char, int> m;
        for(char x: s)
            m[x]++;
        for(char x: t){
            m[x]--;
            if(m[x] == 0)
                m.erase(x);
        }
        if(m.size()==0)
            return true;
        else
            return false;
        
    }
};

File: leetcode/242_easy_valid-anagram.py
class Solution(object):
        def isAnagram(self, s, t):
            """
            :type s: str
            :type t: str
            :rtype: bool
            """
            if len(s) != len(t):
                 return False
            count_dict = dict()
            for elem in s:
                if elem in count_dict.keys():
                    count_dict[elem] +=1
                else:
                    count_dict[elem] = 1
            for elem in t:
                if elem in count_dict.keys():
                    count_dict[elem] -=1
                else:
                    return False
            return all([val == 0 for val in count_dict.values()])